packages/@ourworldindata/utils/src/DetailsOnDemand/detailsOnDemand.tsx

```diff
@@ -47,29 +47,29 @@ export class DoDWrapper extends React.Component<DoDWrapperProps> {
     constructor(props: DoDWrapperProps) {
         super(props)
     }
-    @computed get category() {
+    @computed get category(): string {
         return this.props.category
     }
-    @computed get term() {
+    @computed get term(): string {
         return this.props.term
     }
-    @computed get detail() {
+    @computed get detail(): Detail | void {
```

Comment:
While valid, is there any reason why we're using `void` over `undefined`? (here and above)

Reply:
The Generator case requires it - it throws an error without `void`

In this case, there's functionally not much of a difference between undefined and void seeing as it's being used in a union. I wrote it because I had `void` in my mind from the above example, but I guess `undefined` makes slightly more sense as we should endeavour to use void only [in cases like the ones the docs mention](https://www.typescriptlang.org/docs/handbook/2/functions.html#return-type-void).